networkit/test/test_distance.py

```diff
@@ -99,53 +99,94 @@ def eucledianDist(u):
 		testMesh(9, 18)
 		testMesh(7, 1)
 	
-	def testDiameter(self):
+	def testAdamicAdarDistance(self):
+		for g in self.genERGraphs():
+			g.indexEdges()
+			adam = nk.distance.AdamicAdarDistance(g)
+			adam.preprocess()
+			self.assertIsInstance(adam.getAttribute(),list)
+			e = nk.graphtools.randomEdge(g)
+			self.assertIsInstance(adam.distance(e[0],e[1]),float)
+
+	def testAlgebraicDistance(self):
+		for g in self.genERGraphs():
+			g.indexEdges()
+			alge = nk.distance.AlgebraicDistance(g, withEdgeScores=True)
+			alge.preprocess()
+			self.assertIsInstance(alge.getEdgeScores(),list)
+			e = nk.graphtools.randomEdge(g)
+			self.assertIsInstance(alge.distance(e[0],e[1]),float)			
+		
+	def testCommuteTimeDistance(self):
+		for g in self.genERGraphs():
+			if not g.isDirected():
+				g.indexEdges()
+				ctd = nk.distance.CommuteTimeDistance(g)
+				ctd.runApproximation()
+				ctd.runParallelApproximation()
+				ctd.runSinglePair(nk.graphtools.randomNode(g),nk.graphtools.randomNode(g))
+				ctd.runSingleSource(nk.graphtools.randomNode(g))
```

Comment:
I suggest testing each algorithm in different tests and verify that the produced results are correct.